rest/route/github.go

```diff
@@ -230,43 +227,72 @@ func (gh *githubHookApi) Run(ctx context.Context) gimlet.Responder {
 		}
 
 	case *github.MergeGroupEvent:
-		var msg string
-		if disableMergeGroup {
-			msg = "merge group received, skipping"
-		} else {
-			msg = "merge group received, attempting to queue"
-		}
-		grip.Info(message.Fields{
+		return gh.handleMergeGroupEvent(event)
+	}
+
+	return gimlet.NewJSONResponse(struct{}{})
+}
+
+func (gh *githubHookApi) handleMergeGroupEvent(event *github.MergeGroupEvent) gimlet.Responder {
+	org := event.GetOrg().GetLogin()
+	repo := event.GetRepo().GetName()
+	branch := strings.TrimPrefix(event.MergeGroup.GetBaseRef(), "refs/heads/")
+	grip.Info(message.Fields{
+		"source":   "GitHub hook",
+		"msg_id":   gh.msgID,
+		"event":    gh.eventType,
+		"org":      org,
+		"repo":     repo,
+		"base_sha": event.GetMergeGroup().GetBaseSHA(),
+		"head_sha": event.GetMergeGroup().GetHeadSHA(),
+		"message":  "merge group received",
+	})
+	ref, err := model.FindOneProjectRefWithCommitQueueByOwnerRepoAndBranch(org, repo, branch)
+	if err != nil {
+		grip.Error(message.WrapError(err, message.Fields{
 			"source":   "GitHub hook",
 			"msg_id":   gh.msgID,
 			"event":    gh.eventType,
-			"org":      event.GetOrg().GetLogin(),
-			"repo":     event.GetRepo().GetName(),
+			"org":      org,
+			"repo":     repo,
 			"base_sha": event.GetMergeGroup().GetBaseSHA(),
 			"head_sha": event.GetMergeGroup().GetHeadSHA(),
-			"message":  msg,
+			"message":  "finding project ref",
+		}))
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "finding project ref"))
+	}
+	if ref == nil {
+		grip.Error(message.Fields{
+			"source":   "GitHub hook",
+			"msg_id":   gh.msgID,
+			"event":    gh.eventType,
+			"org":      org,
+			"repo":     repo,
+			"base_sha": event.GetMergeGroup().GetBaseSHA(),
+			"head_sha": event.GetMergeGroup().GetHeadSHA(),
+			"message":  "no matching project ref",
 		})
-
-		// This will be removed when EVG-19964 is ready.
-		if disableMergeGroup {
-			return gimlet.NewJSONResponse(struct{}{})
-		}
-		if err := gh.AddIntentForGithubMerge(event); err != nil {
-			grip.Error(message.WrapError(err, message.Fields{
-				"source":   "GitHub hook",
-				"msg_id":   gh.msgID,
-				"event":    gh.eventType,
-				"org":      event.GetOrg().GetLogin(),
-				"repo":     event.GetRepo().GetName(),
-				"base_sha": event.GetMergeGroup().GetBaseSHA(),
-				"head_sha": event.GetMergeGroup().GetHeadSHA(),
-				"message":  "can't add intent",
-			}))
-			return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "adding patch intent"))
-		}
+		return gimlet.NewJSONInternalErrorResponse(errors.Wrap(err, "no matching project ref"))
 	}
-
-	return gimlet.NewJSONResponse(struct{}{})
+	if ref.CommitQueue.MergeQueue == model.MergeQueueGitHub {
+		err = gh.AddIntentForGithubMerge(event)
+	} else {
+		return gimlet.NewJSONResponse(struct{}{})
```

Comment:
What does this use case look like? What buttons would a user click to get them into a state where we are getting MergeGroupEvents but it's not enabled? I'm guessing they enabled it on github but not on evergreen? 
This might be something to think about at a later time, but I'm wondering if there's something we should do to provide feedback to the user letting them know that there is one more step they need to do to complete the setup.

Reply:
I think there might be an edge case where a user is using both Evergreen and another CI system. If they prefer to use the other CI system for the merge queue, Evergreen would still receive merge group events, since it would be registered for web hooks, but it shouldn't build them.